Makefile

```diff
@@ -99,36 +99,64 @@ doc_R_pkg = \
 
 depends = .doc/$(1) .install/$(1) .check/$(1) .test/$(1)
 
-
 ### Rules
 
 .PHONY: all install check test document shiny \
             check_base check_models check_modules 
 
-all: install document
-
+all: install document ## Install all packages and generate documentation
 
-check_base: $(BASE_C) 
-check_models: $(MODELS_C) 
+check_base: $(BASE_C)     ## Check base packages  
+check_models: $(MODELS_C) ## Check model packages   
 
 # Install base first as Modules has a circular dependency on base,
 # and then run a check on modules
-check_modules: $(BASE_I) $(MODULES_C) 
+check_modules: $(BASE_I) $(MODULES_C) ## Install base packages, check module packages
+
+.PHONY: document
+document: ## Generate documentation for all packages or a single package: make document path/to-package
+	@if [ "$(filter-out $@,$(MAKECMDGOALS))" ]; then \
+		PKG="$(filter-out $@,$(MAKECMDGOALS))"; \
+		echo "Documenting package: $$PKG"; \
+		$(call doc_R_pkg, $$PKG); \
+	else \
+		echo "Documenting all packages"; \
+		$(MAKE) $(ALL_PKGS_D) .doc/base/all; \
+	fi
 
-document: $(ALL_PKGS_D) .doc/base/all
-install: $(ALL_PKGS_I) .install/base/all
-check: $(ALL_PKGS_C) .check/base/all
-test: $(ALL_PKGS_T) .test/base/all
-shiny: $(SHINY_I)
+install: $(ALL_PKGS_I) .install/base/all ## Install all packages
+check: $(ALL_PKGS_C) .check/base/all     ## Check all packages
+test: $(ALL_PKGS_T) .test/base/all       ## Test all packages
+shiny: $(SHINY_I)                        ## Install Shiny app dependencies
 
-# Render the PEcAn bookdown documentation
-book:
-	cd ./book_source && make build
+
+book: 
+	cd ./book_source && make build ## Render the PEcAn bookdown documentation
 
 # Make the timestamp directories if they don't exist yet
 .doc .install .check .test .shiny_depends $(call depends,base) $(call depends,models) $(call depends,modules):
 	mkdir -p $@
 
+### Help
```

Comment:
Nit: move up two spots to land between `shiny` and `book`,  so that all the phony rules come before all the non-phony ones